tests/core_tests/test_context.py

```diff
@@ -153,8 +156,9 @@ def test_additional_entities_equal(self):
             {"type": "Sequence", "id": 456, "bar": "foo"},
         ]
         context_2 = context.Context(self.tk, **kws2)
-        self.assertTrue(context_1 == context_2)
-        self.assertFalse(context_1 != context_2)
+        self.assertEqual(context_1, context_2)
+        # Assert that hashing function treats these as unequal
+        self.assertNotEqual(hash(context_1), hash(context_2))
```

Comment:
Why should the hashes be not equal, when the contexts are marked as equal and the method name states they should be equal?
I assume it is because the additional entities are different so it produces a different has, but we still want to consider the context's the same when not hashing?

Reply:
I think we should probably err on the side of caution here and make then unequal. Could you put a comment in there saying why the result is different even if the contexts are equal?

It's one of those feature in Toolkit that I think we've talked about killing at some point, since it's so poorly supported anywhere anyway.